paper/src/main/java/com/github/igotyou/FactoryMod/inputItem/InputItemMap.java

```diff
@@ -0,0 +1,243 @@
+package com.github.igotyou.FactoryMod.inputItem;
+
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.Repairable;
+import vg.civcraft.mc.civmodcore.inventory.items.ItemMap;
+import vg.civcraft.mc.civmodcore.inventory.items.ItemUtils;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+//////////////////////////////////
+// The class is not thread-safe //
+//////////////////////////////////
+public class InputItemMap {
```

Comment:
I know we've already go the code here but it might be worth considering for the future to move this implementation inside of CivModCore to enhance ItemMap, I'm sure we could always use tag/list parsing in other places.